emcc.py

```diff
@@ -2483,6 +2496,8 @@ def consume_arg_file():
         # difference).
         shared.Settings.DEBUG_LEVEL = 3
     elif check_flag('-profiling') or check_flag('--profiling'):
+      if compile_only:
+        exit_with_error('-profiling is a link time setting! (Pass e.g. -g2 at compile stage instead)')
       shared.Settings.DEBUG_LEVEL = max(shared.Settings.DEBUG_LEVEL, 2)
```

Comment:
I wonder if we should have two different DEBUG_LEVEL settings to avoid this confusion?

Reply:
i.e. LINK_DEBUG_LEVEL and COMPILE_DEBUG_LEVEL?

Reply:
Presumably two different settings are needed if was want to make these two commands do different things:

`emcc hello.c --profiling`   (doesn't include debug info from hello.c)

`emcc hello.c --profiling -g`    (does include debug info from hello.c)

Alternatively we should delay processing of the linker flags until after the compilation is done?